src/components/App.jsx

```diff
@@ -2,18 +2,16 @@ import React from "react";
 import config from "../config";
 
 export default React.createClass({
-  componentDidUpdate() {
-    const scrollTop = (window.pageYOffset || document.documentElement.scrollTop)  - (document.documentElement.clientTop || 0);
 
-    if (location.hash.length > 0 && scrollTop < 500) {
-      try {
-        const searchState = JSON.parse(decodeURIComponent(location.hash.replace(/^#q=/, "")));
-        window.scrollTo(0, searchState.scrollTop);
-      } catch (e) {
-        console.log(e);
-      }
-    }
+  shouldComponentUpdate(nextProps, nextState) {
+    return nextProps.location.pathname !== this.props.location.pathname;
```

Comment:
If I read this correctly then the only way to change a app *or a subcomponent of app* is when the path of the url changes.